### DISCUSSION TITLE
Adding new shard to existing ClickHouse cluster is not working

### DISCUSSION THREAD

**Discussion by adjiekrazz** [akvorado/akvorado#1342] [Q&A] [answered]
Hi everyone. Today I tried to implement ClickHouse clustering with Akvorado and tried to add a new shard to an existing ClickHouse cluster (previously 2s 1r) without deleting the existing data. I think the new shard was successfully connected with the other instance and can communicate well. But I don't know why the new instance failed to migrate its tables. Here are some logs I've collected:

```
logs in orchestrator:
2024-08-02 10:08:15 {"level":"error","error":"code: 36, message: An error occurred in [newshard.xxx:9000]: Code: 36. DB::Exception: Dictionary (`networks`) not found. (BAD_ARGUMENTS) (version 23.8.16.40 (official build))","time":"2024-08-02T03:08:15Z","caller":"akvorado/orchestrator/clickhouse/networks.go:180","module":"akvorado/orchestrator/clickhouse","message":"failed to refresh network dictionary"}

logs on clickhouse-server.err.log:
2024.08.02 03:07:58.198024 [ 258 ] {} <Error> default.flows.DirectoryMonitor.default: Code: 60. DB::Exception: Received from newshard.xxx:9000. DB::Exception: Table default.flows_local does not exist.
```

Then when I check the Kafka UI, the ClickHouse consumer count is not increase (the spare partition is ready). Does anyone know about this issue? or maybe I can do a self-migration of the table?

Thanks

**Comment by vincentbernat** [comment, marked as answer]
It looks like the tables are not created automatically when you add the shard. Tables are created/modified with the "ON CLUSTER" and I suppose it is only done on existing shards. You can check the existing table with `SHOW TABLES` and copy their definition `SHOW CREATE TABLE networks` when missing.

**Reply by adjiekrazz** [reply]
It works! Thank you for the answer.
